### ISSUE TITLE
External data still broken after #22527

### ISSUE THREAD

**Issue by Felixoid** [ClickHouse/ClickHouse#23648]
**Describe the bug**
After #22527 the post of external data still broken when there is only one line. When there are more then one, `\r` is appended to the latest.

**Does it reproduce on recent release?**
It's reproducible in 21.4.5.46
[The list of releases](https://github.com/ClickHouse/ClickHouse/blob/master/utils/list-versions/version_date.tsv)

**How to reproduce**
```
docker run --rm --net=host --name=clickhouse yandex/clickhouse-server:21.4.5.46
```

```
$ printf 'string\nstring\n' | curl -F 'metrics_list=@-;' 'http://localhost:8123/?metrics_list_format=TSV&metrics_list_structure=Path+String&query=SELECT+*+FROM+metrics_list'
string
string
$ printf 'string\nstring' | curl -F 'metrics_list=@-;' 'http://localhost:8123/?metrics_list_format=TSV&metrics_list_structure=Path+String&query=SELECT+*+FROM+metrics_list'
string
string\r
$ printf 'string\n' | curl -F 'metrics_list=@-;' 'http://localhost:8123/?metrics_list_format=TSV&metrics_list_structure=Path+String&query=SELECT+*+FROM+metrics_list'
string
$ printf 'string' | curl -F 'metrics_list=@-;' 'http://localhost:8123/?metrics_list_format=TSV&metrics_list_structure=Path+String&query=SELECT+*+FROM+metrics_list'
Code: 117, e.displayText() = DB::Exception:
You have carriage return (\r, 0x0D, ASCII 13) at end of first row.
It's like your input data has DOS/Windows style line separators, that are illegal in TabSeparated format. You must transform your file to Unix format.
But if you really need carriage return at end of string value of last column, you need to escape it as \r.: (at row 1)

Row 1:
Column 0,   name: Path, type: String, parsed text: "string<CARRIAGE RETURN>"

: While executing SourceFromInputStream (version 21.4.5.46 (official build))
```

**Expected behavior**
I expect the first two and the last two commands would return the same output.

**Comment by filimonov** [comment]
Seems related to multipart parsing.

**Comment by filimonov** [comment]
21.1 & 21.2 return correct result

**Comment by Felixoid** [comment]
Yes, it works in 21.1 and 21.2, but broken in 21.3 and 21.4

**Comment by alexey-milovidov** [comment]
No one guarantees that you can write TSV without a newline character on the last line.
That's why I change the category from "bug" to "compatibility".

**Comment by abyss7** [comment]
@Felixoid can you check v21.7? - I can't reproduce it.

**Comment by Felixoid** [comment]
21.7 is not available on docker hub, unfortunately

But it's still there for 21.6

```
for str in 'string\nstring\n' 'string\nstring' 'string\n' 'string'; do echo test $str:; printf $str curl -F 'metrics_list=@-;' 'http://localhost:8123/?metrics_list_format=TSV&metrics_list_structure=Path+String&query=SELECT+*+FROM+metrics_list'; done
test string
string
:
string
string
test string
string:
string
stringtest string
:
string
test string:
string(no EOL here)
```

**Comment by evillique** [comment]
After some tests it turned out this issue was resolved in #24399. This fix wasn't yet included with 21.4.5.46 and some other versions, that's why you get this error.

**Comment by alexey-milovidov** [comment]
@evillique Let's add the tests from this issue.
